### PR TITLE
[AWSApiModule] support default GovCloud STS endpoint

### PR DESCRIPTION
**Pull request by michal-dagan**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes:[link to the issue](https://jira-dc.paloaltonetworks.com/browse/XSUP-40392)

## Description
Support default GovCloud STS endpoint

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by bziser — APPROVED** [review]

File: Packs/ApiModules/Scripts/AWSApiModule/AWSApiModule.py [hunk 1]
```diff
@@ -106,6 +111,7 @@ def aws_session(self, service, region=None, role_arn=None, role_session_name=Non
             kwargs.update({'Policy': self.aws_role_policy})
 
         demisto.debug(f'{kwargs=}')
+        self.sts_endpoint_url = self.sts_endpoint_url or STS_ENDPOINTS.get(region) or STS_ENDPOINTS.get(self.aws_default_region)
```
Comment (bziser) [inline Packs/ApiModules/Scripts/AWSApiModule/AWSApiModule.py:114]: please add a test

File: Packs/AWS-ACM/ReleaseNotes/1_1_38.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### AWS - ACM
+
+- Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```
Comment (ShirleyDenkberg) [inline Packs/AWS-ACM/ReleaseNotes/1_1_38.md:6]:
```suggestion
Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```

File: Packs/AWS-AccessAnalyzer/ReleaseNotes/1_1_35.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### AWS - AccessAnalyzer
+
+- Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```
Comment (ShirleyDenkberg) [inline Packs/AWS-AccessAnalyzer/ReleaseNotes/1_1_35.md:6]:
```suggestion
Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```

File: Packs/AWS-Athena/ReleaseNotes/2_0_3.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### AWS - Athena
+
+- Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```
Comment (ShirleyDenkberg) [inline Packs/AWS-Athena/ReleaseNotes/2_0_3.md:6]:
```suggestion
Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```

File: Packs/AWS-CloudWatchLogs/ReleaseNotes/1_2_24.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### AWS - CloudWatchLogs
+
+- Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```
Comment (ShirleyDenkberg) [inline Packs/AWS-CloudWatchLogs/ReleaseNotes/1_2_24.md:6]:
```suggestion
Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```

File: Packs/AWS-EC2/ReleaseNotes/1_4_14.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### AWS - EC2
+
+- Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```
Comment (ShirleyDenkberg) [inline Packs/AWS-EC2/ReleaseNotes/1_4_14.md:6]:
```suggestion
Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```

File: Packs/AWS-EKS/ReleaseNotes/1_0_5.md [hunk 7]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### AWS-EKS
+
+- Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```
Comment (ShirleyDenkberg) [inline Packs/AWS-EKS/ReleaseNotes/1_0_5.md:6]:
```suggestion
Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```

File: Packs/AWS-GuardDuty/ReleaseNotes/1_3_53.md [hunk 8]
```diff
@@ -0,0 +1,10 @@
+
+#### Integrations
+
+##### AWS - GuardDuty
+
+- Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
+
+##### AWS - GuardDuty Event Collector
+
+- Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```
Comment (ShirleyDenkberg) [inline Packs/AWS-GuardDuty/ReleaseNotes/1_3_53.md:10]:
```suggestion
Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.

##### AWS - GuardDuty Event Collector

Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```

File: Packs/AWS-IAM/ReleaseNotes/1_1_66.md [hunk 9]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### AWS - Identity and Access Management
+
+- Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```
Comment (ShirleyDenkberg) [inline Packs/AWS-IAM/ReleaseNotes/1_1_66.md:6]:
```suggestion
Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```

File: Packs/AWS-IAMIdentityCenter/ReleaseNotes/1_0_5.md [hunk 10]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### AWS - IAM Identity Center
+
+- Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```
Comment (ShirleyDenkberg) [inline Packs/AWS-IAMIdentityCenter/ReleaseNotes/1_0_5.md:6]:
```suggestion
Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```

File: Packs/AWS-Lambda/ReleaseNotes/1_3_8.md [hunk 11]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### AWS - Lambda
+
+- Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```
Comment (ShirleyDenkberg) [inline Packs/AWS-Lambda/ReleaseNotes/1_3_8.md:6]:
```suggestion
Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```

File: Packs/AWS-Organizations/ReleaseNotes/1_0_4.md [hunk 12]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### AWS - Organizations
+
+- Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```
Comment (ShirleyDenkberg) [inline Packs/AWS-Organizations/ReleaseNotes/1_0_4.md:6]:
```suggestion
Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```

File: Packs/AWS-Route53/ReleaseNotes/1_1_36.md [hunk 13]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### AWS - Route53
+
+- Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```
Comment (ShirleyDenkberg) [inline Packs/AWS-Route53/ReleaseNotes/1_1_36.md:6]:
```suggestion
Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```

File: Packs/AWS-S3/ReleaseNotes/1_2_27.md [hunk 14]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### AWS - S3
+
+- Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```
Comment (ShirleyDenkberg) [inline Packs/AWS-S3/ReleaseNotes/1_2_27.md:6]:
```suggestion
Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```

File: Packs/AWS-SNS/ReleaseNotes/1_0_18.md [hunk 15]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### AWS - SNS
+
+- Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```
Comment (ShirleyDenkberg) [inline Packs/AWS-SNS/ReleaseNotes/1_0_18.md:6]:
```suggestion
Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```

File: Packs/AWS-SQS/ReleaseNotes/1_2_28.md [hunk 16]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### AWS - SQS
+
+- Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```
Comment (ShirleyDenkberg) [inline Packs/AWS-SQS/ReleaseNotes/1_2_28.md:6]:
```suggestion
Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```

File: Packs/AWS-SecurityHub/ReleaseNotes/1_3_36.md [hunk 17]
```diff
@@ -0,0 +1,10 @@
+
+#### Integrations
+
+##### AWS - Security Hub
+
+- Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
+
+##### AWS Security Hub Event Collector
+
+- Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```
Comment (ShirleyDenkberg) [inline Packs/AWS-SecurityHub/ReleaseNotes/1_3_36.md:10]:
```suggestion
Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.

##### AWS Security Hub Event Collector

Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```

File: Packs/AWS-SecurityLake/ReleaseNotes/1_0_11.md [hunk 18]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Amazon Security Lake
+
+- Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```
Comment (ShirleyDenkberg) [inline Packs/AWS-SecurityLake/ReleaseNotes/1_0_11.md:6]:
```suggestion
Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```

File: Packs/AWS_SystemManager/ReleaseNotes/1_0_5.md [hunk 19]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### AWS - System Manager
+
+- Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```
Comment (ShirleyDenkberg) [inline Packs/AWS_SystemManager/ReleaseNotes/1_0_5.md:6]:
```suggestion
Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```

File: Packs/AWS_WAF/ReleaseNotes/1_0_9.md [hunk 20]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### AWS-WAF
+
+- Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```
Comment (ShirleyDenkberg) [inline Packs/AWS_WAF/ReleaseNotes/1_0_9.md:6]:
```suggestion
Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```

File: Packs/Aws-SecretsManager/ReleaseNotes/1_0_43.md [hunk 21]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Aws Secrets Manager
+
+- Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```
Comment (ShirleyDenkberg) [inline Packs/Aws-SecretsManager/ReleaseNotes/1_0_43.md:6]:
```suggestion
Fixed an issue where, when using an AWS GovCloud account, the STS endpoint was not applied correctly if the *STS PrivateLink URL* parameter was not provided.
```

**Comment by ShirleyDenkberg** [comment]
@bziser Doc review completed.

**Pull request merged**